### ISSUE TITLE
Discussion point: master terminology

### ISSUE THREAD

**Issue by paulclarkin** [git/git-scm.com#1487]
I've had a look but can't see this already discussed - apologies if it has.

With Microsoft/GitHub recently discussing the removal of terms such as "master/slave"/"whitelist/blacklist" from their systems, including Github, could or should the term be review and/or removed from the Git reference pages and replaced with a term such as "main"?

I appreciate this would be a large task so might have to be done in stages.

(I've also raised this for discussion on The Pro Git book issue list https://github.com/progit/progit2/issues/1462)

**Comment by jnavila** [comment]
The git reference pages are managed with the git project. This project is only a publishing site for them. For genuine content, we would be happy to review PRs.

**Comment by peff** [comment]
Yes, what @jnavila said.

Also, the term "master" is generally used in conjunction with the default branch name. Any documentation changes would need to be connected to a change in the code. There have been several threads (and some patches) on the mailing list over the past 6 weeks or so. So if you're interested in participating, the first step would probably be to look through the archive to catch up. (The word "slave" doesn't appear at all in Git's documentation, and only in the code as a method call to perl's IO::Pty, which is outside of our control).

The whitelist/blacklist phrases _do_ appear in the documentation, and shouldn't be that hard to change, as there are no code or functionality dependencies. I don't think they've gotten much attention upstream, so any patches to the upstream project would be welcome there.

**Comment by AlexChesser** [comment]
Further to the above, I'm currently reading the documentation regarding git rebase and it is already quite jarring to read `master` everywhere https://git-scm.com/book/en/v2/Git-Branching-Rebasing it feels like a bit of an anachronism.

I suspect that this will only get worse over time for new developers who grow up in a world where they've never called it that.

Are the images in the documentation generated, or would that be a manual task to take on?

There are quite a few instances where a large orange block has "master" written on it?  The find-and-replace in the rest of the docs doesn't seem like that much of a challenge.  Which I'd be perfectly willing to take on.

Incidentally, not sure if this is a "new issue" type comment, but the `main` branch on this repo is still master.  With github's "virtually-1-click" migration tool (as can be seen in the closing comment of https://github.com/progit/progit2/issues/1462) there's no reason not to do it IMHO.

I know in my day job we renamed the branch in *quite literally* dozens of repos to `main` and the only casualty was one of my git aliases.

**Comment by peff** [comment]
@AlexChesser Any changes to the book content should be proposed in the progit2 repo. My understanding is that the images there are done manually (and when updated, there's a semi-manual process to actually update our imports of them for the site, so please do open an issue or PR here if they get updated in the book upstream).

As for whether it's a good idea to change the book content yet, I'm not sure. The default in Git is still `master`, so IMHO it probably make sense for the book to follow that.

As to _this_ repository, I have no objection to changing the default branch. It's actually been on my todo list for a while; the big question is just what if anything might break in our workflows.

**Comment by HonkingGoose** [comment]
@AlexChesser The ProGit2 book repository already has a issue to discuss changing `master` -> `main` in the book itself, see https://github.com/progit/progit2/issues/1462. Feel free to add your comments there, so we can discuss it in the right place. 😉

**Comment by HonkingGoose** [comment]
## Migrating to `main`

> As to _this_ repository, I have no objection to changing the default branch. It's actually been on my todo list for a while; the big question is just what if anything might break in our workflows.

### Read the `github/renaming` readme first

@peff I recommend you read the README at https://github.com/github/renaming before starting your migration. 😉 

### Main things you need to change manually

The main things you need to change yourself:

- Change instances of `master` to `main` in your Travis CI/GitHub Actions configuration files
- Fix-up bot configurations (stale bots/etc...)
- Rake build scripts that have hard-coded references to `master`
- Other scripts that rely on a hard-coded references to a branch

### Rough checklist

1. Create a PR that changes the instances of `master` -> `main` where needed (CI/bot config/rake scripts/shell scripts/etc).
2. Migrate to `main` via the GitHub web interface.
3. Merge the PR that changes the instances of `master` -> `main` to get all your CI to work again.
4. Pin a message on the repository to help others fix up their forks.

---

# Example pinned message for repo:

The `git/git-scm` repo has renamed its default branch from `master` to `main`.
If you don't have a fork yet, you can stop reading now.

If your fork still uses `master` as the default branch name, you will need to change that to `main` as well. Follow the instructions below:

## Steps to rename your fork's default branch

It's easy to fix your fork, as you can do most of the work straight from the GitHub web interface.

1. Go to your forks "homepage" on GitHub.
2. GitHub will have a popup over your `master` branch name, to tell you that the upstream repository has changed its default branch name.
3. Rename your forks branch from `master` to `main` from within the GitHub interface (Settings -> Branches -> Default branch -> click on the pencil icon to rename).
4. Let GitHub automatically rename your fork's branch name.
5. Go back to your fork's homepage, and follow the instructions GitHub gives you to rename the branch name on your local clone of your fork.

## Copy of GitHub's instructions to rename the branch on your local clone

```git
git branch -m master main
git fetch origin
git branch -u origin/main main
```

**Comment by peff** [comment]
@HonkingGoose Thanks for that link. Aside from fixing a bunch of unrelated problems with the repo, it was pretty painless to switch; see #1596.

**Comment by HonkingGoose** [comment]
Hi @peff Do we still need this issue _here_?

There's an issue to update the references in the ProGit2 book, and the content of the Git manual needs to be fixed upstream.
The `git-scm` repository now uses a `main` branch so that's fixed on our part as well.

I did a search on the live `git-scm` website (https://git-scm.com/search/results?search=master) and the only hits for `master` are those in upstream documents that we don't control directly.

**Comment by peff** [comment]
Nope, I think we can close.  Thanks!
